src/walkway.js

```diff
@@ -324,8 +328,10 @@
   function getPolylineLength(polyline) {
     var dist = 0;
     var x1, x2, y1, y2;
+    var i;
```

Comment:
You are absolutely right. But in my opinion `let` is being used without any concern.
You could replace `var` with `let` inside a `for` loop but pulling it out of the loop will still be a bit more performant.
I know you can not tell the difference in performance impact because this will be a few milliseconds but I try to write good, fast and easy maintainable code 😄